lib/trento/application/integration/discovery/policies/host_policy.ex

```diff
@@ -151,6 +151,12 @@ defmodule Trento.Integration.Discovery.HostPolicy do
          admin_username: admin_username
        }
 
+  defp parse_cloud_provider_metadata(
+         :aws,
+         aws_metadata
+       ),
+       do: aws_metadata |> Map.from_struct()
```

Comment:
Question: what kind of struct is `aws_metadata`?

I would expect a similar behavior between the two variants of the funcion
`parse_cloud_provider_metadata(:azure, map)` and `parse_cloud_provider_metadata(:aws, map)`

Reply:
It has the same exact fields used in the `AwsProvider`. We don't really need any modification.
If you want, I can match the fields one by one.
I will do that. I guess that it will help a bit even though the code is more verbose

Reply:
@nelsonkopliku Made more verbose in: https://github.com/trento-project/web/pull/648/commits/26aa4e7e1aafca8dbfab1488801263aef91714d7

Having the next snippet would make the code a bit more meaningful, but well, not worth spending more time on this hehe

```
defp parse_cloud_provider_metadata(
         :aws,
         %AwsMetadata{} = aws_metadata
       ),
       do: Map.from_struct(aws_metadata)
```